test/http.cpp

```diff
@@ -244,6 +252,33 @@ void test_response_parsing() {
     throw std::runtime_error("Response parsing failed");
 }
 
+void test_cors_preflight() {
+  // a server who lets us snoop on what its doing
+  zmq::context_t context;
+  testable_http_server_t server(context, "ipc:///tmp/test_http_server",
+                                "ipc:///tmp/test_http_proxy_upstream", "ipc:///tmp/test_http_results",
+                                "ipc:///tmp/test_http_interrupt");
+  server.passify();
+
+  // get a preflight request as a zmq message that we can enqueue to the server
+  http_request_t request;
+  request.headers.emplace("Access-Control-Request-Method", "POST");
+  request.headers.emplace("Access-Control-Request-Headers", "origin");
+  request.headers.emplace("Origin", "https://foo.bar");
+  request.method = OPTIONS;
+  request.body = "/baz";
+  request.version = "HTTP/1.1";
+  auto req_str = request.to_string();
+
+  // send the request
+  http_request_t request_state;
+  server.enqueue("", req_str, request_state);
+
+  // we dont have preflight support implemented yet so we expect nothing to happen here
+  if (server.last_responses.size() != 0)
+    throw std::logic_error("Cors preflight is not implemented but we got a response");
```

Comment:
@ldsmoreira once we get this merged to master and then merged into your branch, you can change this bit here to check that the shortcircuiter replied properly here